plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -377,49 +393,54 @@ private void setColumnStatistics(String objectName, List<ColumnStatisticsObj> st
     public List<String> getPartitionNames(String databaseName, String tableName)
             throws TException
     {
-        return client.getPartitionNames(databaseName, tableName, (short) -1);
+        return client.getPartitionNames(prependCatalogToDbName(catalogName, databaseName), tableName, (short) -1);
     }
 
     @Override
     public List<String> getPartitionNamesFiltered(String databaseName, String tableName, List<String> partitionValues)
             throws TException
     {
-        return client.getPartitionNamesPs(databaseName, tableName, partitionValues, (short) -1);
+        return client.getPartitionNamesPs(prependCatalogToDbName(catalogName, databaseName), tableName, partitionValues, (short) -1);
     }
 
     @Override
     public int addPartitions(List<Partition> newPartitions)
             throws TException
     {
+        if (newPartitions != null) {
+            for (Partition p : newPartitions) {
+                p.setCatName(catalogName);
```

Comment:
We shouldn't modify the pararameters sent to ThriftHiveMetastoreClient.
Consider creating new corresponding instances internally within the logic of the methods of the ThriftHiveMetastoreClient if the catalog name needs to be set.